### ISSUE TITLE
Scale control

### ISSUE THREAD

**Issue by 1ec5** [mapbox/mapbox-gl-native#1278]
Apple Maps (but not `MKMapView` itself) shows a scale (in both miles and kilometers) in the top-left corner of the map view while you pinch to zoom in or out. A scale would be a useful feature to bake into `MGLMapView`, either off by default or off except while you pinch. It would help to give context, especially since GL doesn’t snap the zoom level to increments and allows very high zoom levels.

**Comment by 1ec5** [comment]
Correction: `MKMapView.showsScale = YES` turns on a persistent scale at the top-left corner.

**Comment by Bluezen** [comment]
Hi @1ec5 ! Do you know if there is a plan to add this feature in an upcoming release?

If not, how would you recommend me to proceed with the current iOS SDK 3.2.3 if I wanted to show a simple scale ? 
Adding a custom UIView on top of the MKMapView which, on every Map position changes (`-mapViewRegionIsChanging:` `-mapView:regionDidChangeAnimated:` ), uses `visibleCoordinateBounds` (or something else ?) to extrapolate and update its display?

**Comment by 1ec5** [comment]
We haven't scheduled this feature into an upcoming release yet, but we aren't opposed to getting the feature in if someone steps up with an implementation.

In the meantime, you have the right idea with the delegate methods (although you might find `-mapViewDidFinishRenderingFrame:fullyRendered:` to be more predictable). In terms of calculating the width of the scale's indicators, I'd use `-[MGLMapView metersPerPointAtLatitude:]`, passing in the current center latitude.

If you come up with an implementation that you feel would be appropriate for the SDK, we'd be happy to consider your PR.

**Comment by Bluezen** [comment]
Thanks for pointing out `-[MGLMapView metersPerPointAtLatitude:]` and `mapViewDidFinishRenderingFrame:fullyRendered` !

Okay, if I am satisfied with my implementation I will propose a PR.

**Comment by friedbunny** [comment]
Somewhat relatedly, here’s a quick way to get the geographic distance spanned by the map view:

```swift
func mapView(mapView: MGLMapView, regionDidChangeAnimated animated: Bool) {
     let screenWidthInPoints = Double(mapView.bounds.size.width)
     let metersPerPoint = mapView.metersPerPointAtLatitude(mapView.centerCoordinate.latitude)
     let screenDistanceInMeters = Int(screenWidthInPoints * metersPerPoint)

     print("\(screenWidthInPoints) × \(metersPerPoint) = \(screenDistanceInMeters)m")
}
```

```
// Portrait iPhone 6++++
414.0 × 4.87247101190374 = 2017m
// Rotate to landscape
736.0 × 4.87247101190374 = 3586m
```

**Comment by 1ec5** [comment]
This is GL JS parity issue now that mapbox/mapbox-gl-js#2940 has landed.

**Comment by 1ec5** [comment]
An Android equivalent is being implemented in #6555.

**Comment by andrewstay** [comment]
Could you please update on the progress on this feature?

**Comment by frederoni** [comment]
Work towards this issue is happening in #7432

**Comment by 1ec5** [comment]
The scale bar is being implemented in #7631.
